### DISCUSSION TITLE
Porous jump boundary condition

### DISCUSSION THREAD

**Discussion by MTODC** [nextfoam/baram#94] [Q&A]
Hello,
I set up a simple model composed by a circular duct and an inside surface (of the same dimensions of the orthogonal section of the duct) representing a dissipative media, modeled by the porous jump boundary condition. 

Link for the model: https://drive.google.com/file/d/16Rfagts_KE2NRcC6nB5DHybWq3nLi65P/view?usp=drive_link

The mesh was done according to #74 , i.e. by using a single surface in the middle of the duct and selecting only the "interface" condition in BaramMesh; in order to avoid the known issue reported in #84 , the mesh and the flow calculation were perfomed in single-core. Despite this, the model isn't able to run: initially the Console reported the error in the image. 

![image](https://github.com/user-attachments/assets/707e4145-50f3-43bd-aed9-bc3077169a85)

I modified the 'matchTolerance' setting in createPatchDict to 1E-2 and this error disappeared, but now after few second that the calulation is started, it terminates. No errors are in the Console. 
I also tried with multi-processor mesh and calculation and I had a similar bheaviour, i.e. termination of calculation without errors in the Console.

Thank you in advance for your help.

Best regards

**Comment by jiban** [comment]
Hi, @MTODC.
Thank you for the detailed report.
Current BaramMesh has an issue in handling interface if the interface is used as a cyclic.
It will be fixed in next release coming soon.

By the way, I have a question.
You mentioned *createPatchDict* in your comment.
I wonder what *createPatchDict* you are talking about.
Could you tell me more about it?

Best regards,
Jake

**Comment by MTODC** [comment]
Sure, I am referring to createPatchDict file that i found in the installation folder of Baram, i.e. in BARAM\solvers\openfoam\etc\caseDicts\annotated. In the file, you can see that is commented a proper section regarding this specific issue and it is recommended to increase tolerance. I refer to many threads on CFD-online ([for example](https://www.cfd-online.com/Forums/openfoam-pre-processing/241557-matching-error-cyclic-patches-spite-having-same-number-faces.html)); this problem seems to arise frequently in multi-core, however I have it also in single core.

Thank you for the support and I'm looking for the next release of Baram!

Best regards

**Comment by torodecamargue** [comment]
Hello,

You can solve this problem in writing a line in each patch in the boundary file (in constant - polyMesh) 

![image](https://github.com/user-attachments/assets/95f7123b-8153-4c94-a6a4-ca7783515d0f)

matchTolerance 2;

Julien

**Comment by torodecamargue** [comment]
Maybe in the next release, it could help if the default settings have a bigger tolerance for the porous jump.

100% of my cases I have to increase manually the tolerance in the constant boundary file.

Best regards

Julien

**Comment by MTODC** [comment]
Hello @torodecamargue ,
I confirm that your method works, I have finally been able to run my model (only in single-core). I thought that the createPatchDict file in BARAM\solvers\openfoam\etc\caseDicts\annotated was a global statement, but it didn't work as well. 
Thank you very much!

Best regards

**Comment by MTODC** [comment]
Instead of using the porous jump condition, I tried to implement a porous region, as reported in the tutorial section of the Baram website, with negligible thickness. I had no issues to run the model in multi-core and I noticed a more accurate results respect to the analytical data.
